File: DecompiledGameScripts/core/gameplay/prereqs/puppet/NPCRarityPrereq.swift

public class NPCRarityPrereq extends IScriptablePrereq {

  public let m_rarity: gamedataNPCRarity;

  public let m_invert: Bool;

  protected func Initialize(recordID: TweakDBID) -> Void {
    let str: String = TweakDBInterface.GetString(recordID + t".rarity", "");
    this.m_rarity = IntEnum<gamedataNPCRarity>(Cast<Int32>(EnumValueFromString("gamedataNPCRarity", str)));
    this.m_invert = TweakDBInterface.GetBool(recordID + t".invert", false);
  }

  protected const func IsOnRegisterSupported() -> Bool {
    return false;
  }

  protected const func OnApplied(state: ref<PrereqState>, game: GameInstance, context: ref<IScriptable>) -> Void {
    state.OnChanged(this.IsFulfilled(game, context));
  }

  public const func IsFulfilled(game: GameInstance, context: ref<IScriptable>) -> Bool {
    let owner: ref<GameObject> = context as GameObject;
    let targetPuppet: ref<ScriptedPuppet> = owner as ScriptedPuppet;
    let rarity: gamedataNPCRarity = targetPuppet.GetNPCRarity();
    if NotEquals(rarity, this.m_rarity) {
      return this.m_invert ? true : false;
    };
    return this.m_invert ? false : true;
  }
}

File: Definitions/Game_Definitions/classes/NPCRarityPrereq.lua
---@meta
---@diagnostic disable

---@class NPCRarityPrereq : gameIScriptablePrereq
---@field rarity gamedataNPCRarity
---@field invert Bool
NPCRarityPrereq = {}

---@return NPCRarityPrereq
function NPCRarityPrereq.new() return end

---@param props table
---@return NPCRarityPrereq
function NPCRarityPrereq.new(props) return end

---@param recordID TweakDBID|string
function NPCRarityPrereq:Initialize(recordID) return end

---@param context IScriptable
---@return Bool
function NPCRarityPrereq:IsFulfilled(context) return end

---@return Bool
function NPCRarityPrereq:IsOnRegisterSupported() return end

---@param state gamePrereqState
---@param context IScriptable
function NPCRarityPrereq:OnApplied(state, context) return end

